### PR TITLE
Oidc auth

### PR DESCRIPTION
**Pull request by pengfei99**
Hi guys

Sorry for the delay. I added the oidc token auth support

Have a nice day

### CODE REVIEW

File: atlasclient/client.py [hunk 1]
```diff
@@ -72,10 +72,17 @@ def __init__(self, host, port=None, username=None, password=None,
         if identifier is None:
             identifier = 'python-atlasclient'
 
-        self.client = HttpClient(host=self.base_url, username=username,
-                                 password=password, identifier=identifier,
-                                 validate_ssl=validate_ssl, timeout=timeout,
-                                 max_retries=max_retries, auth=auth)
+        if username is not None and password is not None and oidc_token is None:
```
Comment (verdan) [inline atlasclient/client.py]: @pengfei99 I believe you don't need to make this check here. simply pass all the arguments to `HttpClient` in this case.

File: atlasclient/client.py [hunk 2]
```diff
@@ -116,16 +123,25 @@ class HttpClient(object):
...
                  timeout=10, max_retries=5, auth=None):
-        basic_token = utils.generate_http_basic_token(username=username, password=password)
-        self.request_params = {
-            'headers': {'X-Requested-By': identifier,
-                        'Authorization': 'Basic {}'.format(basic_token)},
-            # 'auth': (username, password),
-            'verify': validate_ssl,
-            'timeout': timeout,
-        }
+        if username is not None and password is not None and oidc_token is None:
```
Comment (verdan) [inline atlasclient/client.py]: the checks here can be simplified. can you please try to make it simplified?
```python
        if oidc_token:
            auth_header = f'Bearer {oidc_token}'
        elif username and password:
            basic_token = utils.generate_http_basic_token(username=username, password=password)
            auth_header = f'Basic {basic_token}'
        else:
            raise BadHttpAuthArg
        self.request_params = {
            'headers': {'X-Requested-By': identifier,
                        'Authorization': auth_header},
            'verify': validate_ssl,
            'timeout': timeout,
        }
```

**Review by verdan — CHANGES_REQUESTED** [review]
@pengfei99 a couple of small remarks. and I'll be happy to merge this.

**Comment by pengfei99** [comment]
Thanks for the correction, I will apply the changes and do a new PR

**Pull request closed without merging**